godel_process_planning/src/trajectory_utils.cpp

```diff
@@ -0,0 +1,101 @@
+#include "trajectory_utils.h"
+
+
+//////////////////////////////////////////////
+// Helper Functions for joint interpolation //
+//////////////////////////////////////////////
+
+static unsigned int calculateRequiredSteps(const std::vector<double>& start,
+                                           const std::vector<double>& stop,
+                                           double dtheta)
+{
+  // calculate max steps required
+  unsigned steps = 0;
+  for (std::size_t i = 0; i < start.size() ; ++i)
+  {
+    unsigned this_joint_steps = static_cast<unsigned>(std::abs(start[i] - stop[i]) / dtheta);
+    steps = std::max(steps, this_joint_steps);
+  }
+
+  return steps;
+}
+
+static std::vector<double> calculateJointSteps(const std::vector<double>& start,
+                                               const std::vector<double>& stop,
+                                               unsigned int steps)
+{
+  // Given max steps, calculate delta for each joint
+  std::vector<double> result;
+  result.reserve(start.size());
+
+  for (std::size_t i = 0; i < start.size(); ++i)
+  {
+    result.push_back((stop[i] - start[i]) / steps);
+  }
+
+  return result;
+}
+
+static std::vector<double> interpolateJointSteps(const std::vector<double>& start,
+                                                 const std::vector<double>& step_size,
+                                                 unsigned step)
+{
+    std::vector<double> result;
+    result.reserve(start.size());
+    for (std::size_t i = 0; i < start.size(); ++i)
+        result.push_back( start[i] + step_size[i] * step );
+    return result;
+}
+
+
+
+godel_process_planning::PoseVector
+godel_process_planning::interpolateCartesian(const Eigen::Affine3d &start,
+                                             const Eigen::Affine3d &stop,
+                                             double ds)
```

Comment:
This function specifies a maximum cartesian speed, but makes no effort to check rotational speed. This is not a problem with the types of trajectories it handles now, but could be a problem in the future.